pkg/queue/consumers/kafka/consumer.go

```diff
@@ -0,0 +1,96 @@
+package kafka
+
+import (
+	"fmt"
+	"time"
+
+	"gopkg.in/confluentinc/confluent-kafka-go.v1/kafka"
+)
+
+const (
+	POLL_INTERVAL = 1
+)
+
+type Consumer struct {
+	consumer           *kafka.Consumer
+	UncommitedMessages []string
+}
+
+type ConsumerConfig struct {
+	BrokerList string
+	GroupID    string
+	Topic      string
+}
+
+func NewConsumer(config interface{}) *Consumer {
```

Comment:
How will it help if we have a generic Newclient - because at least one level above we should know what we are invoiking  - producer/consumer kafka/pulsar etc